### ISSUE TITLE
activeDeadlineSeconds is setted greater than original ,the value is invalid.

### ISSUE THREAD

**Issue by xingfeng1234** [kubernetes/kubernetes#90553]
**What happened**:
First,you can create job like below yaml file
```
apiVersion: batch/v1
  kind: Job
  metadata:
    name: job-test-deadline
    namespace: default
  spec:
    completions: 1
    parallelism: 1
    activeDeadlineSeconds: 40
    template:
      metadata:
        creationTimestamp: null
      spec:
        containers:
        - command:
          - bin/bash
          - -c
          - sleep 120s
          image: *.*.*.*:30012/library/tomcat:8.5.39
          imagePullPolicy: IfNotPresent
          name: container0
          resources: {}
        restartPolicy: Never
```

Then, edit the job, set the parameter activeDeadlineSeconds greater than 40. You will find that activeDeadlineSeconds is reached, but the job is still running.
**What you expected to happen**:
ActiveDeadlineSeconds is reached the time which is updated, job can be stopped.

**How to reproduce it (as minimally and precisely as possible)**:
you can reproduce it, follow the steps above.

**Anything else we need to know?**:
nothing

**Environment**:
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.6", GitCommit:"96fac5cd13a5dc064f7d9f4f23030a6aeface6cc", GitTreeState:"clean", BuildDate:"2019-08-19T11:05:16Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.6", GitCommit:"96fac5cd13a5dc064f7d9f4f23030a6aeface6cc", GitTreeState:"clean", BuildDate:"2019-08-19T11:05:16Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}

```

**Comment by Joseph-Irving** [comment]
I can reproduce this behaviour on Kubernetes 1.18.0

**Comment by Joseph-Irving** [comment]
I think I've found the root cause of this and I've raised a PR with a potential fix https://github.com/kubernetes/kubernetes/pull/90728

**Comment by xingfeng1234** [comment]
> I think I've found the root cause of this and I've raised a PR with a potential fix #90728

The next step, what should I do? Close the question?

**Comment by Joseph-Irving** [comment]
Once a fix is merged into master, this issue will automatically get closed.
